torch/distributed/device_mesh.py

```diff
@@ -203,6 +203,7 @@ def __init__(
                 if isinstance(mesh, torch.Tensor)
                 else torch.tensor(mesh, dtype=torch.int)
             )
+            assert mesh.device.type == "cpu", f"`mesh` must be a CPU tensor, got {mesh}"
```

Comment:
> Simple fix to assert only for tensor mesh
> 
> AttributeError: 'list' object has no attribute 'device'

We had a discussion above and we want to raise a ValueError, since this is user facing. 

Quoting @awgu: We can prefer to raise real errors (i.e. not assert ...) if it is user facing. Practically for PyTorch, assert cond, msg and if not cond: raise AssertionError(msg) are the same, but distinguishing them can be one way to indicate what is an internal assert vs. intended to be user-facing.